sqlmesh/core/engine_adapter/athena.py

```diff
@@ -0,0 +1,441 @@
+from __future__ import annotations
+from functools import lru_cache
+import typing as t
+import logging
+from sqlglot import exp
+from sqlmesh.core.dialect import to_schema
+from sqlmesh.core.engine_adapter.mixins import PandasNativeFetchDFSupportMixin
+from sqlmesh.core.engine_adapter.trino import TrinoEngineAdapter
+from sqlmesh.core.node import IntervalUnit
+import os
+from sqlmesh.utils.errors import SQLMeshError
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_fixed
+from sqlmesh.core.engine_adapter.shared import (
+    CatalogSupport,
+    DataObject,
+    DataObjectType,
+    CommentCreationTable,
+    CommentCreationView,
+)
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import SchemaName, TableName
+
+logger = logging.getLogger(__name__)
+
+
+def _ensure_valid_location(value: str) -> str:
+    if not value.startswith("s3://"):
```

Comment:
Shouldn't this be validated in the connection config instead?

Reply:
It's also used in `_table_location` to ensure that final location is valid since that is assembled at runtime and relies on user input from `physical_properties`

Reply:
I see. I'd still add validation to the config to fail faster.

Reply:
Ok, i've moved it to a common location and call it from a Pydantic validator as well as within the Athena adapter. I also added some tests to show it was working